subsys/net/lib/config/CMakeLists.txt

```diff
@@ -14,4 +14,6 @@ if(CONFIG_NET_CONFIG_SETTINGS)
     )
 endif()
 
-zephyr_library_sources_ifdef(CONFIG_NET_CONFIG_CLOCK_SNTP_INIT init_clock_sntp.c)
+zephyr_include_directories(.)
```

Comment:
Internal headers shouldn't be exposed globally, whatever needs to be exposed for the app should be placed in a public header.